### PR TITLE
[WFLY-14692] Tests security domain with dots in the name

### PR DESCRIPTION
**Pull request by dcihak**
Creates security domain with a name containing dots and tries to log in using the POST method through the secured form.

Test case for [WFCORE-4707](https://issues.redhat.com/browse/WFCORE-4707).

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/security/SecurityDomainDotNameServerSetupTask.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
+package org.jboss.as.test.integration.management.api.security;
```
Comment (darranl) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/security/SecurityDomainDotNameServerSetupTask.java:1]: Missing copyright header.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/security/SecurityDomainDotNameTestCase.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
+package org.jboss.as.test.integration.management.api.security;
```
Comment (darranl) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/security/SecurityDomainDotNameTestCase.java:1]: Missing copyright header.

**Review by darranl — CHANGES_REQUESTED** [review]
As a submission to the WildFly project this should be using a WFLY Jira issue to track the changes even if the component is just testsuite.

This test is testing deprecated code, when we remove legacy security we are more likely to just delete all legacy security tests.  

But the underlying issue sounds to be very generic, is this reproducible using non-deprecated code as this test will likely be deleted again within the next 12 months.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/security/SecurityDomainDotNameTestCase.java (hunk 2 above)
Comment (dcihak) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/security/SecurityDomainDotNameTestCase.java:23]: Copyright added.

**Comment by dcihak** [comment]
@darranl Thanks for the review!
JIRA issue [WFLY-14692](https://issues.redhat.com/browse/WFLY-14692) was created to be linked with this test case. The original issue was related to the security subsystem and it seems it is not reproducible with the non-legacy configuration. As we need this test case for an authomatic verification of the 7.3.7 JBEAP issue [JBEAP-21215](https://issues.redhat.com/browse/JBEAP-21215) (will be merged into EAP AS TS) do you think it will be better to merge this into  wildfly-core testsuite?

**Comment by darranl** [comment]
In general my preference is get the test case as close as possible to the code it is testing, from an engineers perspective it makes it faster for an engineer to identify if they have caused a regression in that code rather than waiting for a later testsuite to detect the regression.

Having said that this test does require the legacy security subsystem for it's reproducer and the life of the test upstream is potentially limited so I probably would not worry at this stage.

Do be aware once we reach WildFly 25 or 26 development we may be ready to pass the point where we stop accepting legacy security changes and tests upstream so at some point in the future these kind of tests may not need an upstream change although we are not quite there yet.

**Comment by dcihak** [comment]
Retest this please.

**Comment by dcihak** [comment]
@darranl We are ok with the limited life of the test case, but thanks for the information. I fixed the test for the case of Galleon JDK 11. Do we need some other changes to be done from my side? Thanks.

**Comment by dcihak** [comment]
@darranl Hello, are there any changes required from your side or can we merge this? We would like to use this test case for the 7.3.7 JIRA verificaton which is currently in progress. All the checks have passed. Thanks.

**Review by darranl — APPROVED** [review]

**Pull request merged**